### DISCUSSION TITLE
Export of percentage_charged

### DISCUSSION THREAD

**Discussion by steveauch** [netzero-labs/netzero#13] [Q&A]
Great product @zigam, love your work. 
I just did my first export from the app and was surprised to see that percentage_charged wasn't one of the exported fields. Any reason for this, or am I missing something?

**Comment by zigam** [comment]
Thanks for the feedback!  Battery SoC is not currently exported, the data is in a slightly different format so it requires a bit of work to include it.  I'll add this to the list.

In the meantime: if you're comfortable running some Python code, you can get the data (and a much easier way to export bulk data) with this tool:
https://github.com/netzero-labs/tesla-solar-download

**Reply by eJonny** [reply]
Am I reading the Tesla-solar-download readme properly? It outputs two files per day:  power (in 5 minute increments) and energy (in 24 hour increments). How would I derive battery state of charge unless there were an energy_left value with some timestamp (which appears can be pulled from the live_status end point). I can't seem to find a state of charge value in any other energy end point, which is odd.

POWER
solar_power,
battery_power,
grid_power,
load_power

ENERGY
solar_energy_exported,
grid_energy_imported,
grid_energy_exported_from_solar,
grid_energy_exported_from_battery,
battery_energy_exported,
battery_energy_imported_from_grid,
battery_energy_imported_from_solar,
consumer_energy_imported_from_grid,
consumer_energy_imported_from_solar,
consumer_energy_imported_from_battery

**Reply by zigam** [reply]
I need to update the documentation, but there are additional `soe` files with 15 min intervals that only output the state of charge.

**Reply by chrisj7903** [reply]
Add my vote for SoC to be included in NetZero downloads !!
 
@eJonny - to derive SoC from 5-min Power readings is not straight forward, but here you go:
Because their are 12 x 5mins in an hour you divide each Power reading for the Powerwall by 12 to get the KWh discharged or charged in each 5 minute period. (If the 5 min readings are in Watts, divide by 1000 as well to get kWh).
Then if you add them all up you will know many kWh have been consumed by (or provided to) the Powerwall over any period of time you choose. 
A Powerwall2 holds 13.5kWh at 100% SoC I believe, so making the above calcs (starting from a time when the battery was 100% charged) will reveal the kWh remaining, and thus SoC is simply KWh remaining / 13.5 * 100
